aws/src/main/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystem.java

```diff
@@ -293,21 +298,16 @@ public Message readFile(S3Object f, String charset) throws FileSystemException,
 
 	/**
 	 * Attempts to resolve the Local S3 Pointer created by the {@link #toFile(String) toFile} method.
-	 * Or returns directly when the provided S3Object is an actual representation of the S3 bucket
+	 * Updates the Metadata context but does not retrieve the actual file handle.
 	 */
-	private S3Object resolve(S3Object f) {
-		if(f instanceof LocalS3Object) {
-			return s3Client.getObject(bucketName, f.getKey());
+	private S3Object updateFileAttributes(S3Object f) {
+		if(f.getObjectMetadata().getRawMetadataValue(Headers.CONTENT_LENGTH) == null) {
+			ObjectMetadata omd = s3Client.getObjectMetadata(bucketName, f.getKey());
+			f.setObjectMetadata(omd);
 		}
 		return f;
```

Comment:
Do you actually still need the return value of these methods in these file systems?

Reply:
Not really ":)